tasks/test-tutorial.js

```diff
@@ -0,0 +1,97 @@
+import fs from 'fs'
+import os from 'os'
+import path from 'path'
+
+import execa from 'execa'
+
+const testTutorial = async () => {
+  // First two args are "node" and path to script
+  const [, , pathToProject] = process.argv
+
+  let projectPath = pathToProject
+  const frameworkPath = path.join(__dirname, '..')
+  const e2ePath = path.join(frameworkPath, 'tasks/e2e')
+
+  if (process.env.DEBUG) {
+    console.log(`📁 ~ file: test-tutorial.js ~ projectPath`, projectPath)
+    console.log(`🌲 ~ file: test-tutorial.js ~ frameworkPath`, frameworkPath)
+    console.log(`🚀 ~ file: test-tutorial.js ~ e2ePath`, e2ePath)
+  }
+
+  await execa('yarn install', {
+    cwd: e2ePath,
+    shell: true,
+    stdio: 'inherit',
+  })
+
+  if (pathToProject) {
+    console.log(
+      `\n 🗂️  You have supplied a path ${projectPath}, we will not create a new `
+    )
+    console.log('Redwood project, we will use the app you have specified.')
+  } else {
+    console.log('\n ℹ️  You have not supplied a path to a Redwood project.')
+    console.log('We will create one for you. \n \n')
+    console.log(
+      "📋 We will copy './packages/create-redwood-app/template' and link packages/* \n"
+    )
+
+    // Use temporary project path, because no user supplied one
+    projectPath = fs.mkdtempSync(path.join(os.tmpdir(), 'redwood-e2e-'))
+
+    await execa(
+      'yarn babel-node',
+      ['src/create-redwood-app.js', projectPath, '--no-yarn-install'],
+      {
+        cwd: path.join(frameworkPath, 'packages/create-redwood-app'),
+        shell: true,
+        stdio: 'inherit',
+      }
+    )
+  }
+
+  // Clean and build framework
+  await execa('yarn build:clean && yarn lerna run build:js', {
+    cwd: frameworkPath,
+    shell: true,
+    stdio: 'inherit',
+  })
+
+  const packagesPath = path.join(frameworkPath, 'packages')
+
+  // Link packages from framework
+  await execa(`ln -s ${packagesPath} ./packages`, {
```

Comment:
As a Windows user this looks scary. I'd feel safer using https://nodejs.org/api/fs.html#fs_fs_symlinksync_target_path_type

Reply:
This actually works fine (since we tested rwt link) - remember the process is pretty much identical. But I'll change it over anyway